src/components/MainNav/MainNav.js

```diff
@@ -95,6 +95,9 @@ class MainNav extends Component {
         }
       }
     });
+
+    // remove QueryProvider cache to be 100% sure we're starting from a clean slate.
+    this.props.queryClient.removeQueries();
```

Comment:
Why are we doing this in `<MainNav>`? That feels ... random. I mean, I get that it mounts after authentication, but still it's a weird place for this. Would it be better to hook into the other post-login functions in loginServices, or stash this (and other stuff?) in a dedicated `<AuthenticatedSessionInit>` component? Dunno. Just thinkin'.